openprescribing/frontend/models.py

```diff
@@ -448,6 +458,112 @@ def current_version(self):
                 next_version = version.replaced_by
         return version
 
+    @cached_property
+    def dmd_objs(self):
+        '''Return highest level dm+d objects with this presentation's BNF code.
+        '''
+        for model in [VMP, AMP, VMPP, AMPP]:
+            objs = list(model.objects.filter(bnf_code=self.bnf_code))
+            if objs:
+                return objs
+        return []
+
+    @property
+    def dmd_obj_ids(self):
+        return [obj.id for obj in self.dmd_objs]
+
+    @property
+    def dmd_obj_type(self):
+        if not self.dmd_objs:
+            return None
+        else:
+            return type(self.dmd_objs[0])
+
+    def tariff_categories(self):
+        '''Return all tariff categories for this presentation.
+
+        Tariff categories are set at VMPP level, so this doesn't make sense for
+        presentations that correspond to AMPs.
```

Comment:
I think it's just too early in the morning for me, but why not? 

* VMPP: "500mg paracetamol tablets 28 pack" 
* VMP: "500mg paracetamol tablets"
* AMPP: "500mg paracetamol tables 28 pack (Boots)"
* AMP: "500mg paracetamol tablets (Boots)"

Per a conversation you had with Rich a while ago, didn't we decide it's incorrect to consider AMPs to be in the DT in any case?

But if they are, why aren't AMPPs too?

My sense is that DT would only apply to VMPs and VMPPs?

If my morning brain can't cope with it, it's probably a sign that greater brains with less understanding of dm+d might also get stuck here.

Reply:
The difference is that an AMPP belongs to a VMPP, so we can unambiguously match an AMPP against a DT category.

In contrast, an AMP belongs to a VMP, which has one or more VMPPs, which may be in different DT categories.

Does this make sense?